addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/WebCacheDeceptionScanRule.java

```diff
@@ -0,0 +1,187 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.text.similarity.LevenshteinDistance;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+
+/**
+ * Active scan rule which attempts various 403 bypass techniques.
+ *
+ * @author Nikhil (@bettercalln1ck)
+ */
+public class WebCacheDeceptionScanRule extends AbstractAppPlugin {
+
+    private static final int LEVENSHTEIN_THRESHOLD = 200;
+
+    private static final String MESSAGE_PREFIX = "ascanalpha.webCacheDeception.";
+    private static final Logger log = LogManager.getLogger(WebCacheDeceptionScanRule.class);
+
+    protected static final String[] TEST_EXTENSIONS = {
+        "css", "jpg", "js", "html", "gif", "png", "svg", "php", "txt", "pdf", "jsp", "asp"
+    };
+
+    @Override
+    public int getId() {
+        return 40039;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public void scan() {
+        if (initialTest()) {
+            HttpMessage authorisedMessage = getNewMsg();
+            try {
+                sendAndReceive(authorisedMessage);
+                String originalResponse =
+                        new String(authorisedMessage.getResponseBody().getBytes());
+                URI uri = authorisedMessage.getRequestHeader().getURI();
+                ArrayList<String> extensions = new ArrayList<String>();
+                String path = uri.getPath();
+                // checks whether the page with appended path gets cached or not
+                for (String ext : TEST_EXTENSIONS) {
+                    String newPath = path + "/test." + ext;
+                    uri.setPath(newPath);
+                    authorisedMessage.getRequestHeader().setURI(uri);
+                    sendAndReceive(authorisedMessage);
+                    String method = authorisedMessage.getRequestHeader().getMethod();
```

Comment:
Is there a reason to store this intermediately? Couldn't it just be set inline below? ex:
`                     unauthorisedMessage.getRequestHeader().setMethod(authorisedMessage.getRequestHeader().getMethod());`